lib/db.js

```diff
@@ -302,6 +315,75 @@ class LimitDBRedis extends EventEmitter {
       });
   }
 
+  buildStatusResponse(bucket, prefixLength) {
+    return (key, lastDrip, remaining) => {
+      const bucketKeyConfig = this.bucketKeyConfig(bucket, key);
+      return {
+        key: key.substring(prefixLength),
+        limit: bucketKeyConfig.size,
+        reset: this.calculateReset(bucketKeyConfig, remaining),
+        remaining
+      };
+    };
+  }
+
+  /**
+   * Find all buckets that match the key.
+   *
+   * @param {statusParams} params
+   * @param {function(Error, statusResult)} [callback]
+   */
+  status(params, callback) {
+    const valError = this.validateParams(params);
+    if (valError) {
+      return setImmediate(callback, valError);
+    }
+    const bucket = this.buckets[params.type];
+    let prefix = `${params.type}:${params.key}*`;
```

Comment:
Just a comment I guess you already knew, redis uses this just to skip keys while scanning so it does not has to serialize them but still have to scan through them.